contracts/xchain/PolygonChild.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@fx-portal/contracts/tunnel/FxBaseChildTunnel.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract PolygonChild is FxBaseChildTunnel, Ownable {
+    address public stakeInfoAddress;
+    
+    constructor(address _fxChild) FxBaseChildTunnel(_fxChild) {}
+
+    function _processMessageFromRoot(
+        uint256 /* stateId */,
+        address sender,
+        bytes memory data
+    ) internal override validateSender(sender) {
+        (bool success, /* returnId */ ) = stakeInfoAddress.call(data);
+    }
+
+    function sendMessageToRoot(bytes memory message) public {
+        _sendMessageToRoot(message);
+    }
+
+    function setStakeInfoAddress(address _stakeInfoAddress) public onlyOwner {
+        stakeInfoAddress = _stakeInfoAddress;
+    }
```

Comment:
Can we move this to a constructor and make the contract not Ownable?

Reply:
then we have circular dependencies. PolygonChild will take StakeInfo address in the constructor, and StakeInfo will take PolygonChild address in the constructor (in the updators)

Reply:
Ah good point. Let's keep StakeInfo independent then 👍